src/util/psbt/map/global.rs

```diff
@@ -110,6 +154,55 @@ impl Map for Global {
             });
         }
 
+        // BIP 174: The Combiner must remove any duplicate key-value pairs, in accordance with
+        //          the specification. It can pick arbitrarily when conflicts occur.
+
+        // Keeping the highest version
+        self.version = self.version.max(other.version);
+
+        // Merging xpubs
+        for (xpub, (fingerprint1, derivation1)) in other.xpub {
+            match self.xpub.entry(xpub) {
+                Entry::Vacant(entry) => {
+                    entry.insert((fingerprint1, derivation1));
+                },
+                Entry::Occupied(mut entry) => {
+                    // Here in case of the conflict we select the version with algorithm:
+                    // 1) if fingerprints are equal but derivations are not, we report an error; otherwise
+                    // 2) choose longest unhardened derivation; if equal
+                    // 3) choose longest derivation; if equal
+                    // 4) choose largest fingerprint; if equal
```

Comment:
Largest here means just by ordering the bytes, right? So 0xdeadbeef wins from 0xbeadbeef? This is a bit arbitrary. But I suppose I can agree that it's better to do something then to error.

Reply:
Correct. Actually since fingerprint is `[u8; 4]` and not `u32` I assume it uses little-endian comparison. And yes, this is arbitrary

Reply:
nit: I'd perhaps rephrase "choose largest pingerprint using binary ordering" because this suggests that the fingerprint "size" might have real meaning :p